blueprints/ember-bootstrap/index.js

```diff
@@ -6,60 +6,151 @@ const fs = require('fs');
 const path = require('path');
 const writeFile = rsvp.denodeify(fs.writeFile);
 const chalk = require('chalk');
+const BuildConfigEditor = require('ember-cli-build-config-editor');
+const SilentError = require('silent-error'); // From ember-cli
 
 const bs3Version = '^3.3.7';
 const bs4Version = 'next';
 
+const validPreprocessors = [
+  'none',
+  'less',
+  'sass'
+];
+
 module.exports = {
+  name: 'ember-bootstrap',
+
+  description: 'Configure ember-bootstrap',
+
+  availableOptions: [
+    { name: 'bootstrap-version', type: Number, default: 3, aliases: ['bv'] },
+    { name: 'target-preprocessor', type: String, aliases: ['tp'] }
+  ],
+
+  works: 'insideProject',
+
   normalizeEntityName() {
   },
 
-  afterInstall() {
-    return this.addDependencies()
-      .then(() => this.addPreprocessorImport());
+  beforeInstall(option) {
+    let { bootstrapVersion, targetPreprocessor } = option;
+    if (bootstrapVersion !== 3 && bootstrapVersion !== 4) {
+      throw new SilentError('Bootstrap version must be 3 or 4');
+    }
+
+    if (targetPreprocessor && !validPreprocessors.includes(targetPreprocessor)) {
+      throw new SilentError(`Valid preprocessors are: ${validPreprocessors.join(', ')}`);
+    }
+
+    if (!targetPreprocessor) {
+      let dependencies = this.project.dependencies();
+      if ('ember-cli-sass' in dependencies) {
+        targetPreprocessor = option.targetPreprocessor = 'sass';
+      } else if ('ember-cli-less' in dependencies) {
+        targetPreprocessor = option.targetPreprocessor = 'less';
+      } else {
+        targetPreprocessor = 'none';
+      }
+    }
+
+    if (bootstrapVersion === 4 && targetPreprocessor === 'less') {
+      throw new SilentError('You cannot use LESS with Bootstrap 4');
+    }
+
+    this.ui.writeLine(chalk.green(`Installing for Bootstrap ${bootstrapVersion} using preprocessor ${targetPreprocessor}`));
+  },
+
+  afterInstall(option) {
+    return this.adjustBootstrapDependencies(option)
+      .then(() => this.addPreprocessorStyleImport(option))
+      .then(() => this.addBuildConfiguration(option));
   },
 
-  addDependencies() {
+  adjustBootstrapDependencies(option) {
+    let { bootstrapVersion, targetPreprocessor } = option;
     let dependencies = this.project.dependencies();
+    let bowerDependencies = this.project.bowerDependencies();
+    let promises = [];
 
-    let promises = [
-      this.addPackageToProject('bootstrap', bs4Version)
-    ];
-
-    if ('ember-cli-sass' in dependencies) {
+    if (bootstrapVersion === 4) {
+      if ('bootstrap' in bowerDependencies) {
+        this.ui.writeLine(chalk.white.bgBlue('ACTION: Manually remove \'bootstrap\' from bower.json'));
```

Comment:
It seemed a good example of how we might provide feedback before we invested in removing bower packages. I'll look into the bower removal now.